corehq/util/metrics/__init__.py

```diff
@@ -326,6 +326,17 @@ def limit_domains(domain_name):
     return '__other__'
 
 
+def limit_tags(tags: Dict[str, str], domain: str):
+    from corehq import toggles
+    if toggles.HIGH_COUNT_DETAILED_TAGGING.enabled(domain):
+        return tags
+
+    for key in list(tags.keys()):
+        if key in GATED_DETAILED_TAGS:
+            del tags[key]
+    return tags
```

Comment:
Nit: I'd prefer returning a new dict versus modifying the existing one, just to avoid mutation, which can be tricky to reason about.

```suggestion
    return {k: v for k, v in tags.items() if k not in GATED_DETAILED_TAGS}
```